initrd/bin/kexec-save-default

```diff
@@ -129,6 +129,41 @@ if [ ! -r $ENTRY_FILE -o ! -r $HASH_FILE ]; then
 	die "Failed to write default config"
 fi
 
+# logic to parse OS initrd to get extract crypttab, its filepaths and its OS defined options
+mkdir -p /tmp/initrd_extract
+cd /tmp/initrd_extract
+# User just selected a default boot option. Get initrd filename
+current_default_initrd=$(cat /boot/kexec_default_hashes.txt | grep initr | awk -F " " {'print $NF'} | sed 's/\.\//\/boot\//g')
+# Get crypttab files paths from initrd
+crypttab_files=$(zcat $current_default_initrd | cpio --list --quiet | grep crypttab) > /dev/null || crypttab_files=$(cpio --list --quiet < $current_default_initrd | grep crypttab) > /dev/null
+
+if [ ! -z "$crypttab_files" ]; then
+	echo "Extracting current selected default boot's $current_default_initrd for crypttab analysis..."
+	zcat $current_default_initrd | cpio -id --quiet "$crypttab_files" > /dev/null || cpio -id --quiet < $current_default_initrd "$crypttab_files" > /dev/null
+
+	crypttab_entries=$(cat "$crypttab_files"|uniq| sed 's/none/\/secret.key/g')
+	cd - > /dev/null
+
+	#insert current default boot's initrd crypttab location into tracking file to be overwritten into initramfs
+	echo "The following OS crypttab was found in intrd and Heads will overwrite it: $crypttab_files"
+	echo "Adding found crypptab files location under $bootdir/kexec_initrd_crypttab_path.txt"	
+	echo "$crypttab_files" > "$bootdir/kexec_initrd_crypttab_path.txt"
+else
+	echo "No crypttab file found in extracted initrd. Removing $bootdir/kexec_initrd_crypttab_path.txt"
```

Comment:
I like having this trace and the similar one below, but I'd suggest:
* I don't think we need to say what file we're deleting since it's always the same
* Maybe we should point out that LUKS key sealing will not work since we can't override crypttab

If you had a default boot selected in the past, and are updating it now, is it possible old copies of the files could be hanging around?  Maybe we should `rm -f "$bootdir/kexec_initrd_crypttab_*"` in case an old copy of `..._entries.txt` was afound?

Reply:
@JonathonHall-Purism Same reasoning as before. If for whatever reason a user is keeping backup of /boot into root for backups, I'm sure the user would love to see why there is a new file added, or deleted?

Not sure I understand the reasoning here, I might have missed the point and therefore if that is the case i'm sorry.
Both files are deleted individually in the current PR?

Reply:
> @JonathonHall-Purism Same reasoning as before. If for whatever reason a user is keeping backup of /boot into root for backups, I'm sure the user would love to see why there is a new file added, or deleted?

Yeah agree, thanks for spelling out your thought process here.  Let's keep it :+1: 

> Both files are deleted individually in the current PR?

OK, I see in this case it should go on later to also see that `crypttab_entries` is empty and then delete `..._entries.txt` too.  But per above if you want to handle multiple crypttabs properly, maybe it would be simpler and more robust just to create the initrd override archive here, which would also dodge the issue of keeping two files in sync entirely?